### PR TITLE
Fix/search filter focus

### PR DESCRIPTION
**Pull request by psealock**
Fixes https://github.com/woocommerce/woocommerce-admin/issues/1584

On report pages, the "Show:" dropdown should bring focus to the input if there is one.

### Accessibility

- [x] I've tested using only a keyboard (no mouse)
- [x] I've tested using a screen reader
- [x] All animations respect [`prefers-reduced-motion`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion)
- [x] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)

### Screenshots

focus

### Detailed test instructions:

1. Products Report > Show > Single Product
2. See the focus brought to the input box
3. Click on the "back" button
4. See the focus brought to first element
5. Try again, but this time with a selection. Is focus working?

### CODE REVIEW

**Comment by psealock** [comment]
@Aljullu I was wondering if you could take a look at this one? There is something else stealing focus and I've hit a wall trying to debug. It works fine until there is a product tag already selected. It works sometimes in Chrome (below) and almost never in Firefox.

focus-error

The focus remains either on the `body` or the dropdown button. I've tried multiple approaches with no success. I'm hoping you might have some insight?

**Comment by Aljullu** [comment]
Thanks for working on this tricky issue @psealock. I was testing it and could notice the same issue you reported, but couldn't find what's causing it yet, I will give it another try tomorrow. In any case, what you got so far already handles the most important issue IMO that was happening when navigating inside the dropdown.

**Comment by psealock** [comment]
Thanks for the help @Aljullu! I really appreciate it.

>what you got so far already handles the most important issue IMO that was happening when navigating inside the dropdown

Thats true, we could deploy that portion for now.

**Comment by Aljullu** [comment]
I couldn't find the source of the issue yet, but noticed that replacing [this line of CSS](https://github.com/woocommerce/woocommerce-admin/blob/master/packages/components/src/tag/style.scss#L14) with this:

```CSS
transition: background 0.2s cubic-bezier(0.4, 1, 0.4, 1),
	color 0.2s cubic-bezier(0.4, 1, 0.4, 1);
```

seems to fix the issue:

focus

I don't know exactly what was going on there. Maybe the CSS transition applied to all properties was making the element not to be visible for some ms and that was making the focus not to work?

Let me know if that works for you @psealock!

**Comment by psealock** [comment]
Wow 😲 @Aljullu, nice find here. I had imagined the tag may play a part because it has a popover of its own, but didn't deduce a CSS transition could have played a factor.

I did notice the contents of the dropdown aren't rendered correctly. You can also see the scroll bars in your gif as well.

wierd

I think this may have to do with how the content sizing of the popover are calculated. Its interesting to me how this could be affected by focus.

In any case, a valid point was brought up in the complementary Gutenberg PR:

>It may not be expected / accessible to forcibly "skip" a tabbable element in this way.

In other words, are we running into a11y issues by stealing focus from the first element? Between this comment and the issue seen in the gif, I think its best to limit this PR to directly focusing the input only after the Animation Slider has finished. 

I updated the test instructions to reflect this. How do you feel about this compromise?

Also, many thanks for taking the time to dig into this one!

**Review by Aljullu — APPROVED** [review]
> I had imagined the tag may play a part because it has a popover of its own, but didn't deduce a CSS transition could have played a factor.

It was one of the weirdest issues I have seen in my life, honestly! :smile: 

> I think its best to limit this PR to directly focusing the input only after the Animation Slider has finished.

That sounds good. This PR fixes the main concern I had with the filter and I agree with the point made in the Gutenberg PR.

We can fix it in a follow-up, but I still see the scrollbars issue when there is an active tag:

dropdown

File: packages/components/CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,9 @@
-# (unreleased)
+# 2.0.0 (unreleased)
```
Comment (timmyc) [inline packages/components/CHANGELOG.md]: Nice move on the version bump here with the breaking prop change 👍

**Comment by timmyc** [comment]
🔥 nice PR @psealock
